client/src/reducers/universe.js

```diff
@@ -0,0 +1,65 @@
+// jshint esversion: 6
+
+import { Universe } from "../util/stateManager";
+
+/*
+REST API 0.1 dataframe reducer.
+
+TODO: 0.2 reducer
+*/
+
+const UniverseReducerV01 = (state = new Universe("0.1"), action) => {
```

Comment:
Worth a discussion here. https://redux.js.org/faq/organizingstate#can-i-put-functions-promises-or-other-non-serializable-items-in-my-store-state 

This threw me at first and it took me a minute to find out whether this was something I hadn't done because of my use cases or convention or recommendations or something else. It turned out - see above - plain objects are recommended (and what I would prefer to stick with if possible). I would be interested to discuss converting this to a series of functions and storing the result in redux. If at the end of that conversation this still makes more sense, I'll relent, but I think `return this` is indicating there may be profit / simplicity in a more functional approach.

Reply:
I think you are right - if we want the "pure" (serializable) approach advocated in the FAQ link.   Note that we have the same issue with crossfilter.  While crossfilter is not implemented as an ES6 class, it is not a pure object (it has a prototype with functions).

I need some of your time to understand the right approach.     Universe will be simple to convert to functions & a pure object.     World and crossfilter will be somewhat more complex but possible.

Issues I can see off top of my head:
* Loss of sugar:  all of the `state.world.crossfilter.dosomething()` will become `crossfilterModule.dosomething(state.world.crossfilter)` where `dosomething()` includes the complex getters, setters, etc.
* all embedded objects (crossfilter, KeyValueCache, etc) have to be similarly functionalized, which will increase verbosity.   This affects both modules (stateManager and crossfilter), as they have a lot of embedded non-pure objects.

This will require a significant refactoring of these modules, so I definitely want to understand the requirements.   F2F...